src/main/java/uk/nhs/tis/trainee/notifications/service/NotificationService.java

```diff
@@ -120,24 +158,147 @@ public void execute(JobExecutionContext jobExecutionContext) {
   }
 
   /**
-   * Get the user account details from Cognito (if they have signed-up to TIS Self-Service), or from
-   * the Trainee Details profile if not.
+   * Schedule a notification.
    *
-   * @param personId The person ID to search for.
+   * @param jobId      The job id. This must be unique for programme membership / placement and
+   *                   notification milestone.
+   * @param jobDataMap The map of job data.
+   * @param when       The date to schedule the notification to be sent.
+   * @throws SchedulerException if the job could not be scheduled.
+   */
+  public void scheduleNotification(String jobId, JobDataMap jobDataMap, Date when)
+      throws SchedulerException {
+    JobDetail job = newJob(NotificationService.class)
+        .withIdentity(jobId)
+        .usingJobData(jobDataMap)
+        .storeDurably(false)
+        .build();
+
+    Trigger trigger = newTrigger()
+        .withIdentity(TRIGGER_ID_PREFIX + jobId)
+        .startAt(when)
+        .build();
+
+    Date scheduledDate = scheduler.scheduleJob(job, trigger);
+    log.info("Notification for {} scheduled for {}", jobId, scheduledDate);
+  }
+
+  /**
+   * Remove a scheduled notification if it exists.
+   *
+   * @param jobId The job id key to remove.
+   * @throws SchedulerException if the scheduler failed in its duties (non-existent jobs do not
+   *                            trigger this exception).
+   */
+  public void removeNotification(String jobId) throws SchedulerException {
+    JobKey jobKey = new JobKey(jobId);
+    // Delete the job and unschedule its triggers.
+    // We do not simply remove the trigger, since a replacement job may have different data
+    // (e.g. programme name).
+    scheduler.deleteJob(jobKey);
+    log.info("Removed any stale notification scheduled for {}", jobId);
+  }
+
+  /**
+   * Get a future schedule for a notification from the start date and day offset.
+   *
+   * @param startDate       The starting date.
+   * @param daysBeforeStart The number of days prior to the start date.
+   * @return The notification scheduling date and time.
+   */
+  public Date getScheduleDate(LocalDate startDate, int daysBeforeStart) {
+    Date milestone;
+    LocalDate milestoneDate = startDate.minusDays(daysBeforeStart);
+    if (!milestoneDate.isAfter(LocalDate.now())) {
+      // 'Missed' milestones: schedule to be sent soon, but not immediately
+      // in case of human editing 'jitter'.
+      milestone = Date.from(Instant.now()
+          .plus(PAST_MILESTONE_SCHEDULE_DELAY_HOURS, ChronoUnit.HOURS));
```

Comment:
For testing purposes, you might want to adjust this to ChronoUnit.MINUTES to allow the scheduled notification to be triggered (assuming a placement starting within 12 weeks).